crates/server/src/lib.rs

```diff
@@ -1,3 +1,5 @@
 pub mod api;
+pub mod components;
 pub mod configuration;
+pub mod logic;
```

Comment:
I'm not convinced with this module name here, is there a concept behind the logic here? Are we maybe missing or creating the wrong abstraction?

Reply:
what do you think about "domain" for the module name? basically, it refers to our business logic

Reply:
we want to reuse the existing business logic living in the HTTP controllers in the future RPC Server procedures

Reply:
I think `domain` is a better one for  sure.